src/main/java/uk/gov/companieshouse/api/accounts/service/impl/CreditorsWithinOneYearService.java

```diff
@@ -0,0 +1,134 @@
+import uk.gov.companieshouse.api.accounts.links.TransactionLinkType;
+import uk.gov.companieshouse.api.accounts.model.entity.notes.CreditorsWithinOneYearEntity;
+import uk.gov.companieshouse.api.accounts.model.rest.notes.CreditorsWithinOneYear;
+import uk.gov.companieshouse.api.accounts.repository.CreditorsWithinOneYearRepository;
+import uk.gov.companieshouse.api.accounts.service.ResourceService;
+import uk.gov.companieshouse.api.accounts.service.response.ResponseObject;
+import uk.gov.companieshouse.api.accounts.service.response.ResponseStatus;
+import uk.gov.companieshouse.api.accounts.transaction.Transaction;
+import uk.gov.companieshouse.api.accounts.transformer.CreditorsWithinOneYearTransformer;
+import uk.gov.companieshouse.api.accounts.utility.impl.KeyIdGenerator;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.HashMap;
+import java.util.Map;
+
+import static uk.gov.companieshouse.api.accounts.CompanyAccountsApplication.APPLICATION_NAME_SPACE;
+
+@Service
+public class CreditorsWithinOneYearService implements ResourceService<CreditorsWithinOneYear> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(APPLICATION_NAME_SPACE);
+
+    private CreditorsWithinOneYearRepository repository;
+    private CreditorsWithinOneYearTransformer transformer;
+    private KeyIdGenerator keyIdGenerator;
+    private SmallFullService smallFullService;
+
+    @Autowired
+    public CreditorsWithinOneYearService (CreditorsWithinOneYearRepository repository,
+                                          CreditorsWithinOneYearTransformer transformer,
+                                          KeyIdGenerator keyIdGenerator,
+                                          SmallFullService smallFullService) {
+
+        this.repository = repository;
+        this.transformer = transformer;
+        this.keyIdGenerator = keyIdGenerator;
+        this.smallFullService = smallFullService;
+    }
+
+    @Override
+    public ResponseObject<CreditorsWithinOneYear> create(CreditorsWithinOneYear rest,
+                                                         Transaction transaction,
+                                                         String companyAccountId,
+                                                         HttpServletRequest request) throws DataException {
+
+        setMetadataOnRestObject(rest, transaction, companyAccountId);
+
+        CreditorsWithinOneYearEntity entity = transformer.transform(rest);
+        entity.setId(generateID(companyAccountId));
+
+        try {
+            repository.insert(entity);
+        } catch (DuplicateKeyException e) {
+            LOGGER.errorRequest(request, e, getDebugMap(transaction, companyAccountId, entity.getId()));
+            return  new ResponseObject<>(ResponseStatus.DUPLICATE_KEY_ERROR);
+        } catch (MongoException e) {
```

Comment:
We should wrap the original MongoException here using  
new DataException("Failed to insert "  + ResourceName.CREDITORS_WITHIN_ONE_YEAR.getName(), **me**);

Reply:
not sure what you mean, as I'm pretty sure we are doing that anyway. we wrap the error up and throw a dataException.